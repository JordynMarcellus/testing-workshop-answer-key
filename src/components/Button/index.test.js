import React from "react";
import { Button } from "./";
import { render, fireEvent, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";

describe("Button component", () => {
  afterEach(cleanup);
  it("clicks the button and calls the function", async () => {
    const mockFn = jest.fn();
    const { getByText } = render(
      <Button id="button-test-snapshot" handleClick={mockFn} isDisabled={false}>
        Click me!
      </Button>
    );
    fireEvent.click(getByText("Click me!"));
    expect(getByText("Click me!")).toBeEnabled();
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
  it("does not click a button that is disabled", () => {
    const mockFn = jest.fn();
    const { getByText } = render(
      <Button id="button-test-snapshot" handleClick={mockFn} isDisabled={true}>
        Click me!
      </Button>
    );
    fireEvent.click(getByText("Click me!"));
    expect(getByText("Click me!")).toBeDisabled();
    expect(mockFn).not.toHaveBeenCalled();
  });
  it("renders out the disabled button", () => {
    const mockFn = jest.fn();
    const { container } = render(
      <Button id="button-test-snapshot" handleClick={mockFn} isDisabled={true}>
        Click me!
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("renders out an enabled button", () => {
    const mockFn = jest.fn();
    const { container } = render(
      <Button id="button-test-snapshot" handleClick={mockFn} isDisabled={false}>
        Click me!
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
