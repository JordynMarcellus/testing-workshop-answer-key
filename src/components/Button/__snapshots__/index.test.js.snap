// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders out an enabled button 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  margin: 10px;
  padding: 12px 12px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  text-align: center;
  white-space: nowrap;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  text-transform: none;
  text-transform: capitalize;
  color: #333;
  background-color: #dfe6e9;
  border: 0 none;
  border-radius: 4px;
  font-size: 13px;
  font-weight: 500;
  line-height: 1.3;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 160px;
  -ms-flex: 0 0 160px;
  flex: 0 0 160px;
  box-shadow: 2px 5px 10px var(--color-smoke);
}

.emotion-0:hover {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: 0.85;
}

.emotion-0:active {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: 0.75;
}

.emotion-0:focus {
  outline: 1px dotted #959595;
  outline-offset: -4px;
}

<button
  class="emotion-0 emotion-1"
  data-testid="button-test-snapshot"
  id="button-test-snapshot"
>
  Click me!
</button>
`;

exports[`Button component renders out the disabled button 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  margin: 10px;
  padding: 12px 12px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  text-align: center;
  white-space: nowrap;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  text-transform: none;
  text-transform: capitalize;
  color: #333;
  background-color: #dfe6e9;
  border: 0 none;
  border-radius: 4px;
  font-size: 13px;
  font-weight: 500;
  line-height: 1.3;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 160px;
  -ms-flex: 0 0 160px;
  flex: 0 0 160px;
  box-shadow: 2px 5px 10px var(--color-smoke);
}

.emotion-0:hover {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: 0.85;
}

.emotion-0:active {
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
  opacity: 0.75;
}

.emotion-0:focus {
  outline: 1px dotted #959595;
  outline-offset: -4px;
}

<button
  class="emotion-0 emotion-1"
  data-testid="button-test-snapshot"
  disabled=""
  id="button-test-snapshot"
>
  Click me!
</button>
`;
